swan-lake/development-tutorials/observability/overview-of-ballerina-observability.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Overview of Ballerina Observability
+description: See how Ballerina supports observability by exposing itself via metrics, tracing, and logs to external systems.
+keywords: ballerina, observability, metrics, tracing, logs, opentelemetry
+permalink: /learn/overview-of-ballerina-observability/
+active: overview-of-ballerina-observability
+intro: Observability is a measure of how well the internal states of a system can be inferred from the knowledge of its external outputs.
+---
+
+Ballerina Observability is a built-in feature designed to facilitate monitoring and debugging Ballerina applications by collecting, processing, and analyzing performance and operational data. It integrates seamlessly with observability tools and supports three primary pillars of observability: metrics, tracing, and logging.
+
+1. **Metrics** - Provide quantitative insights into the performance of an application, such as request counts, latencies, and error counts.
+2. **Tracing** - Used to track requests/transactions as they flow through different services in a distributed system from the point of entry to exit.
+3. **Logging** - Text records of activities that occurred with relevant information along with the timestamp with various log levels for further analysis.
+
+Observability is understanding a system's state based on the data it generates, such as logs, metrics, and traces. Observability tools and platforms enable developers to monitor, troubleshoot, and optimize their applications effectively.
+
+Ballerina services and any client connectors are observable by default. HTTP/HTTPS and SQL client connectors use semantic tags to make tracing and metrics monitoring more informative.
+
+## Provide observability in Ballerina
+
+This guide demonstrates how to enable Ballerina observability to observe Ballerina services using different tools and platforms. We will use the example given below.
+
+### Example: A simple Ballerina service to observe
+Create a new Ballerina project using the `bal new` command.
+```
+$ bal new observability_demo
+
+Created new package 'observability_demo' at observability_demo.
+```
+
+Delete everything inside the `main.bal` file which is located inside the newly created `observability_demo` directory, and replace it with the following code to create a Ballerina service.
```

Comment:
Replace the contents of the main.bal file, which is located inside the newly created `observability_demo` directory with the following code:

Reply:
Address review suggestions